airflow_connector.py

```diff
@@ -0,0 +1,101 @@
+"""Connects to the Cloud SQL database."""
+import MySQLdb
+
+
+class AirflowDB(object):
+  """"Provides the methods which allow interaction with the Airflow SQL database."""  # pylint: disable=line-too-long
+
+  def __init__(self, user, password, db, host=None, unix_socket=None):
+    # creating the connection in the object allows for reconnection in event of
+    # a lost connection
+    self._unix_socket = unix_socket
+    self._host = host
+    self._user = user
+    self._password = password
+    self._db = db
+    self._airflow_db = self.create_connection()
+
+  def query(self, request):
+    """Sends an SQL query to the airflow database.
+
+    Args:
+        request: the SQL request (string)
+
+    Returns:
+        The database result (tuple)
+    """
+    # The following ensures that the query executes and returns,
+    # even if the db connection has been lost
+    print request
+    try:
+      cursor = self._airflow_db.cursor()
+      cursor.execute(request)
+      response = cursor.fetchall()
+    except MySQLdb.OperationalError, e:
+      print '*****************ERROR TRIGGERED WITH MYSQLDB*********'
+      print 'Error: '
+      print e
+      print ''
+      if e[0] == 2006:
```

Comment:
duplicating code here,
```python
if e[0] in [2006, 2013]:
```

Reply:
And also use the Error code from the mysql libary ie MYsqldb.<NAME of error code>, instead of just using a number (that has no meaning to me :))